api/src/main/java/org/pmiops/workbench/auth/DelegatedUserCredentials.java

```diff
@@ -0,0 +1,165 @@
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.json.webtoken.JsonWebToken;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.cloud.iam.credentials.v1.IamCredentialsClient;
+import com.google.cloud.iam.credentials.v1.SignJwtRequest;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * OAuth2 Credentials representing a Service Account using domain-wide delegation of authority to
+ * generate access tokens on behalf of a G Suite user.
+ *
+ * <p>This class calls the IAM Credentials API to request a JWT to be signed using a service
+ * account's system-managed private key. This is different from the approach adopted by the
+ * ServiceAccountCredentials class, where an application-provided private key is used to self-sign
+ * the JWT and then exchange it for an access token.
+ *
+ * <p>This use of the IAM Credentials API allows a system to use domain-wide delegation of authority
+ * to authorize calls as end users without loading private keys directly into the application.
+ *
+ * <p>This class shares some patterns in common with the ImpersonatedCredentials class; namely, it
+ * uses the IAM Credentials API to allow one service account to perform some actions on behalf of
+ * another service account. However, this class differs in two notable ways: (1) it supports
+ * impersonation of end users, while ImpersonatedCredentials supports only impersonation of service
+ * accounts, and (2) it relies on application default credentials for simplicity in the All of Us
+ * Researcher Workbench use case.
+ *
+ * <p>Example usage, for authorizing user requests to the Google Directory API:<br>
+ *
+ * <pre>
+ *   DelegatedUserCredentials delegatedCredentials = new DelegatedUserCredentials(
+ *     "service-account-with-dwd-enabled@project-name.iam.gserviceaccount.com",
+ *     "admin-gsuite-user@my-gsuite-domain.com",
+ *     DirectoryScopes.ADMIN_DIRECTORY_USERS);
+ *   Directory directoryClient = new Directory.Builder(
+ *       new NetHttpTransport(), new JacksonFactory(), null)
+ *       .setHttpRequestInitializer(new HttpCredentialsAdapter(delegatedCredentials))
+ *       .build();
+ * </pre>
+ */
+public class DelegatedUserCredentials extends OAuth2Credentials {
+
+  static final String JWT_BEARER_GRANT_TYPE = "urn:ietf:params:oauth:grant-type:jwt-bearer";
+  static final String SERVICE_ACCOUNT_NAME_FORMAT = "projects/-/serviceAccounts/%s";
+  static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  public static final Duration ACCESS_TOKEN_DURATION = Duration.ofMinutes(60);
```

Comment:
This seems high to me. I think 5 minutes might be more typical for this kind of token. Do we have a use case where we'd need a long-lived token here? (and if it expires, the refresh should kick in - right?)